ethcore/src/client/client.rs

```diff
@@ -578,9 +564,49 @@ impl Client {
 		self.db.read().write_buffered(batch);
 		chain.commit();
 		self.update_last_hashes(&parent, hash);
+
+		if let Err(e) = self.prune_ancient(state, &chain) {
```

Comment:
just wondering is this break an implicit precondition that the ancient block should be marked canon before the new block is committed.